cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -162,20 +163,22 @@ private String getSpecJson(Application app, final String bundleVersion, final St
       configurer = artifactId == null ?
         new DefaultAppConfigurer(app, configString) :
         new DefaultAppConfigurer(artifactId, app, configString, artifactRepository, pluginInstantiator);
-    }
 
-    Config appConfig;
-    TypeToken typeToken = TypeToken.of(app.getClass());
-    TypeToken<?> configToken = typeToken.resolveType(Application.class.getTypeParameters()[0]);
-    if (Strings.isNullOrEmpty(configString)) {
-      appConfig = (Config) configToken.getRawType().newInstance();
-    } else {
-      //TODO: CDAP-2869 Handle JsonSyntax Exception better and throw a more meaningful error. This will be done when we
-      //use PluginInstantiator methods instead of GSON deserialization
-      appConfig = GSON.fromJson(configString, configToken.getType());
-    }
+      Config appConfig;
```

Comment:
So it looks like we were just leaving files which are not deleted since we were closing the PluginInstantiator earlier. So functionality was not affected but this change will cleanup the dir.